src/builtins/Transformers.jl

```diff
@@ -513,11 +516,116 @@ MLJBase.output_scitype_union(::Type{<:OneHotEncoder}) = Union{Missing,Found}
+    features::Vector{Symbol}
+    continuous_fill::Union{Function}
+    count_fill::Union{Function}
+    categorical_fill::Union{Function}
+    allowed_scitypes::Array{Any}
+
+end
+
+FillImputer(;features=Symbol[], continuous_fill=median,
+    count_fill=basic_count_fill ,categorical_fill=x->common_cat_filler(x),
+    allowed_scitypes=[Union{Continuous,Missing},Union{Count,Missing},Union{Multiclass,Missing}])= FillImputer(features,
+        continuous_fill,count_fill,categorical_fill,allowed_scitypes)
+
+
+function commonCategoryFiller(vec::CategoricalArray)
+    ftab= freqtable(vec)
+    return names(ftab)[1][findmax(ftab)[2]]
+end
+
+
+function common_cat_filler(vec::CategoricalArray)
+    return StatsBase.mode(vec)
+end
+
+
+
+function basic_count_fill(v)
+    round(eltype(v),median(v))
+end
+
+
+
+mutable struct FillImputerResult <: Unsupervised
+    features::Vector{Symbol}
+end
+
+function fit(transformer::FillImputer,X)
+    all_features = Tables.schema(X).names # a tuple not vector
+    specified_features =isempty(transformer.features) ? collect(all_features) : transformer.features
+
+    features=Symbol[]
+    for j in eachindex(all_features)
+        ftr = all_features[j]
+        col = MLJBase.selectcols(X,j)
+        T = scitype_union(col)
+        if any([T <: a for a in transformer.allowed_scitypes]) && ftr in specified_features
+            push!(features,ftr)
+        end
+
+    end
+
+    fitresult = FillImputerResult(collect(features))
+    report = nothing
+    cache = nothing
+
+    return fitresult, cache, report
+end
+
+
+function transform(transformer::FillImputer, fitresult, X)
+
+    df=DataFrame(X)
+    features = Tables.schema(X).names # tuple not vector
+    issubset(Set(fitresult.features), Set(features) ) ||
+        error("Attempting to transform table with feature labels not seen in fit. ")
+
+    for ftr in fitresult.features
+        mis=ismissing.(df[ftr])
+        mis_ind=findall(x-> ismissing(x),df[ftr])
+        col = MLJBase.selectcols(X,ftr)
+        T = scitype_union(col)
+        if T<:Union{Continuous,Missing}
+            df[mis_ind,ftr]=transformer.continuous_fill(df[:x][map(!,mis)])
+        elseif T <: Union{Count,Missing}
+            df[mis_ind,ftr]=transformer.count_fill(df[:x][map(!,mis)])
+        else
+            df[mis_ind,ftr]=transformer.categorical_fill(df[:x][map(!,mis)])
+        end
+    end
+    df
+end
+MLJBase.load_path(::Type{<:FillImputer}) = "MLJ.FillImputer"
+MLJBase.package_url(::Type{<:FillImputer}) = "https://github.com/alan-turing-institute/MLJ.jl"
+MLJBase.package_name(::Type{<:FillImputer}) = "MLJ"
+MLJBase.package_uuid(::Type{<:FillImputer}) = ""
+MLJBase.is_pure_julia(::Type{<:FillImputer}) = true
+MLJBase.input_scitype_union(::Type{<:FillImputer}) = Union{Missing,Found}
+MLJBase.input_is_multivariate(::Type{<:FillImputer}) = true
+MLJBase.output_scitype_union(::Type{<:FillImputer}) = Union{Missing,Found}
+MLJBase.output_is_multivariate(::Type{<:FillImputer}) = true
+
```

Comment:
Updates to the model API require these changes:

- delete lines 615 and 617; the `input_is_multivariate` and `output_is_multivariate` traits are obsolete.
- replace 614 and 616 with:

```julia
MLJBase.input_scitype(::Type{<:FillImputer}) = MLJBase.Table(Scientific)
MLJBase.ouput_scitype(::Type{<:FillImputer}) = MLJBase.Table(Scientific)
```
Here `Scientific` is just an alias for `Union{Missing,Found}`.